server/chat-plugins/youtube.ts

```diff
@@ -166,9 +185,29 @@ export class YoutubeInterface {
 	save() {
 		return FS(STORAGE_PATH).writeUpdate(() => JSON.stringify(this.data));
 	}
+	setUpdateTimer() {
+		if (this.updateTimer) clearTimeout(this.updateTimer);
+		const time = Date.now();
+		const nextMidnight = new Date(time + 24 * 60 * 60 * 1000);
+		nextMidnight.setHours(0, 0, 1);
+		this.updateTimer = setTimeout(() => this.updateData(), nextMidnight.getTime() - time);
+		return this.updateTimer;
```

Comment:
tbh the return is unnecessary, if you're already setting `this.updateTimer`. Just update the definition to `updateTimer!: NodeJS.Timer`.